Adapt Chromium compiler/linker settings to the OpenBSD system toolchain,
omitting unsupported flags and enabling the wxneeded JIT executable flag.

Index: codex-rs/modcargo-crates/v8-150.4.0/build/config/compiler/BUILD.gn
--- codex-rs/modcargo-crates/v8-150.4.0/build/config/compiler/BUILD.gn.orig
+++ codex-rs/modcargo-crates/v8-150.4.0/build/config/compiler/BUILD.gn
@@ -203,7 +203,7 @@
   # This greatly reduces the size of debug builds, at the cost of
   # debugging information which is required by some specialized
   # debugging tools.
-  simple_template_names = is_clang && !is_win && !is_apple
+  simple_template_names = is_clang && !is_win && !is_apple && !is_bsd
 
   # This switch is used to enable -Wexit-time-destructors by default. This
   # warning serves as a flip switch to allow a gradual migration of targets
@@ -296,7 +296,7 @@
 #
 # Since this is referenced both when passing the file and when deciding which
 # warnings to enable, store it in a separate variable here.
-use_clang_warning_suppression_file =
+use_clang_warning_suppression_file = !is_bsd &&
     is_clang && clang_warning_suppression_file != ""
 
 # default_include_dirs ---------------------------------------------------------
@@ -314,12 +314,15 @@
 # Compiler instrumentation can introduce dependencies in DSOs to symbols in
 # the executable they are loaded into, so they are unresolved at link-time.
 config("no_unresolved_symbols") {
-  if (!using_sanitizer &&
+  if (!using_sanitizer && !is_bsd &&
       (is_linux || is_chromeos || is_android || is_fuchsia)) {
     ldflags = [
       "-Wl,-z,defs",
       "-Wl,--as-needed",
     ]
+  }
+  if (current_cpu == "x86" && is_openbsd) {
+    ldflags = [ "-Wl,-z,notext", "-Wl,--strip-all" ]
   }
 }
 
@@ -440,6 +443,10 @@
     # --------------------------------
     cflags += [ "-fno-strict-aliasing" ]  # See http://crbug.com/32204
 
+    if (is_openbsd) {
+      ldflags += [ "-Wl,-z,wxneeded" ]
+    }
+
     # Stack protection. ShadowCallStack and Stack protector address the same
     # problems. Therefore, we only enable one or the other. Clang advertises SCS
     # as a stronger alternative to StackProtector, so we give SCS precedence
@@ -588,9 +595,9 @@
   if (is_clang) {
     # Flags for diagnostics.
     cflags += [ "-fcolor-diagnostics" ]
-    if (!is_win) {
+    if (!is_win && !is_bsd) {
       cflags += [ "-fdiagnostics-show-inlining-chain" ]
-    } else {
+    } else if (is_win) {
       # Combine after https://github.com/llvm/llvm-project/pull/192241
       cflags += [ "/clang:-fdiagnostics-show-inlining-chain" ]
     }
@@ -615,7 +622,7 @@
     }
   }
 
-  if (is_clang) {
+  if (is_clang && !is_bsd) {
     cflags += [ "-fcrash-diagnostics-dir=" + clang_diagnostic_dir ]
     if (save_reproducers_on_lld_crash && use_lld) {
       ldflags += [
@@ -1071,7 +1078,8 @@
         # errors. We only apply fatal warnings when using lld.
         ldflags += [ "-Wl,-fatal_warnings" ]
       }
-      if (!is_apple && current_os != "aix" && current_os != "zos") {
+      if (!is_apple && !is_openbsd && current_os != "aix" &&
+          current_os != "zos") {
         ldflags += [ "-Wl,--fatal-warnings" ]
       }
     }
@@ -1418,7 +1426,7 @@
       "-file-prefix-map",
       rebase_path("//.", "") + "=" + rebase_path("//.", root_build_dir),
     ]
-    if (!is_win) {
+    if (!is_win && !is_bsd) {
       # We don't use clang -cc1as on Windows (yet? https://crbug.com/762167)
       asmflags = [ "-Wa,-fdebug-compilation-dir,." ]
     }
@@ -1471,7 +1479,7 @@
 
 config("clang_revision") {
   if (is_clang && clang_base_path == default_clang_base_path &&
-      current_os != "zos") {
+      current_os != "zos" && !is_bsd) {
     _perform_consistency_checks = current_toolchain == default_toolchain
     if (llvm_force_head_revision) {
       _head_revision_stamp_path = "//third_party/llvm-build/force_head_revision"
@@ -1584,7 +1592,7 @@
 # See also: https://crbug.com/40891132#comment10
 ubsan_hardening("c_array_bounds") {
   sanitizer = "array-bounds"
-  condition = !(is_asan && target_cpu == "x86")
+  condition = !(is_asan && target_cpu == "x86") && !is_bsd
 
   # Because we've enabled array-bounds sanitizing we also want to suppress
   # the related warning about "unsafe-buffer-usage-in-static-sized-array",
@@ -1598,6 +1606,7 @@
 # `NOTREACHED()` at the end of such functions.
 ubsan_hardening("return") {
   sanitizer = "return"
+  condition = !is_bsd
 }
 
 config("rustc_revision") {
@@ -1930,12 +1939,16 @@
       # TODO(crbug.com/376641662): Fix and re-enable.
       "-Wno-nontrivial-memcall",
 
-      # TODO(crbug.com/432275627): Fix and re-enable.
-      "-Wno-uninitialized-const-pointer",
-
-      # TODO(crbug.com/495753203): Fix and re-enable.
-      "-Wno-unused-but-set-global",
     ]
+
+    if (!is_bsd) {
+      cflags += [
+        # TODO(crbug.com/495753203): Fix and re-enable.
+        "-Wno-unused-but-set-global",
+        # TODO(crbug.com/432275627): Fix and re-enable.
+        "-Wno-uninitialized-const-pointer",
+      ]
+    }
 
     cflags_cc += [
       # TODO(crbug.com/328490295): Fix and re-enable for C flags.
@@ -2304,8 +2317,8 @@
 # 2. Remove the thin_archive config, so that the .a file actually contains all
 #    .o files, instead of just references to .o files in the build directoy
 config("thin_archive") {
-  if ((is_apple && use_lld) || (is_linux && !is_clang) || current_os == "aix" ||
-      current_os == "zos") {
+  if ((is_apple && use_lld) || (is_linux && !is_clang && !is_bsd) ||
+      current_os == "aix" || current_os == "zos") {
     # The macOS and iOS linker ld64.ldd doesn't support thin archive without
     # symbol table, gcc on linux also throws the error `archive has no index`.
     # AIX does support -s option, not -S option.
@@ -2748,7 +2761,7 @@
 # There are some targeted places that AFDO regresses, so we provide a separate
 # config to allow AFDO to be disabled per-target.
 config("afdo") {
-  if (is_clang) {
+  if (is_clang && !is_bsd) {
     cflags = []
     if (clang_emit_debug_info_for_profiling) {
       # Add the following flags to generate debug info for profiling.
@@ -2775,7 +2788,7 @@
       cflags += [ "-Wno-backend-plugin" ]
       inputs = [ _clang_sample_profile ]
     }
-  } else if (auto_profile_path != "" && is_a_target_toolchain) {
+  } else if (auto_profile_path != "" && is_a_target_toolchain && !is_bsd) {
     cflags = [ "-fauto-profile=${auto_profile_path}" ]
     inputs = [ auto_profile_path ]
   }
@@ -2827,10 +2840,6 @@
   # thinlto requires -gsplit-dwarf in ldflags.
   if (use_thin_lto && !is_apple) {
     ldflags = split_dwarf_flags
-  } else {
-    # .dwo files are generated when ThinLTO is not used.
-    c_additional_outputs =
-        [ "{{target_out_dir}}/{{label_name}}/{{source_name_part}}.dwo" ]
   }
 }
 
@@ -2909,7 +2918,7 @@
     }
 
     if (current_os != "zos") {
-      cflags += [ "-g2" ]
+      cflags += [ "-g0" ]
     }
 
     if (is_clang && !is_tsan && !is_asan && !use_dwarf5) {
@@ -2932,7 +2941,7 @@
     # Disable debugger index for Android, which uses lldb. lldb can create (and
     # cache) its own index in about 20 seconds, so there's no need to pay the
     # price of an index on every link.
-    _add_symbol_index = symbol_level == 2 && !is_apple &&
+    _add_symbol_index = symbol_level == 2 && !is_apple && !is_bsd &&
                         (use_lld || use_mold) && !is_wasm && !is_android
     if (_add_symbol_index) {
       if (use_dwarf5) {
@@ -2972,7 +2981,7 @@
     cflags += [ "-gomit-unreferenced-methods" ]
   }
 
-  if (is_clang) {
+  if (is_clang && !is_bsd) {
     if (is_apple) {
       # TODO(crbug.com/40117949): Investigate missing debug info on mac.
       # Make sure we don't use constructor homing on mac.
